src/hotspot/share/runtime/synchronizer.hpp

```diff
@@ -172,29 +123,30 @@ class ObjectSynchronizer : AllStatic {
   // JNI detach support
   static void release_monitors_owned_by_thread(JavaThread* current);
 
+  // Iterate owned ObjectMonitors where the owner `filter` returns true
+  template <typename OwnerFilter>
+  static void monitors_iterate_filtered(MonitorClosure* closure, OwnerFilter filter);
+
   // Iterate ObjectMonitors where the owner == thread; this does NOT include
-  // ObjectMonitors where owner is set to a stack lock address in thread:
-  //
-  // This version of monitors_iterate() works with the in-use monitor list.
+  // ObjectMonitors where owner is set to a stack lock address in thread.
   static void monitors_iterate(MonitorClosure* m, JavaThread* thread);
-  // This version of monitors_iterate() works with the specified linked list.
-  static void monitors_iterate(MonitorClosure* closure,
-                               ObjectMonitorsHashtable::PtrList* list,
-                               JavaThread* thread);
+
+  // Iterate owned ObjectMonitors
+  static void monitors_iterate(MonitorClosure* closure);
```

Comment:
Should the preconditions described in the implementation be documented in the header as well. Or at least have a note along the lines of `See ObjectSynchronizer::monitors_iterate_filtered for details on the contract for using this.`

Reply:
I've unified and updated the comments both files.